mlfinlab/sample_weights/attribution.py

```diff
@@ -0,0 +1,84 @@
+"""
+Logic regarding return and time decay attribution for sample weights from chapter 4.
+"""
+
+import pandas as pd
+import numpy as np
+from mlfinlab.util.multiprocess import mp_pandas_obj
+from mlfinlab.sampling.concurrent import num_concurrent_events, get_av_uniqueness_from_tripple_barrier
+
+
+def _apply_weight_by_return(label_endtime, num_conc_events, close_series, molecule):
+    """
+    Snippet 4.10, page 69, Determination of Sample Weight by Absolute Return Attribution
```

Comment:
For all docstrings: please add a space between the function description and the parameter descriptions.